src/mqueryfront/src/App.js

```diff
@@ -5,42 +5,51 @@ import QueryPage from "./query/QueryPage";
 import RecentPage from "./recent/RecentPage";
 import StatusPage from "./status/StatusPage";
 import ConfigPage from "./config/ConfigPage";
+import AuthPage from "./auth/AuthPage";
 import axios from "axios";
 import { API_URL } from "./config";
 import "./App.css";
 
+function parseJWT(token) {
+    const base64Url = token.split(".")[1];
+    const base64 = base64Url.replace("-", "+").replace("_", "/");
+    return JSON.parse(Buffer.from(base64, "base64").toString("binary"));
+}
+
 function App() {
     const [config, setConfig] = useState(null);
+    const rawToken = localStorage.getItem("token");
+    const token = rawToken ? parseJWT(localStorage.getItem("token")) : null;
 
     useEffect(() => {
         axios.get(`${API_URL}/server`).then((response) => {
             setConfig(response.data);
         });
     }, []);
 
+    const login = (token) => {
```

Comment:
I don't like that name mismatch here.

In upper-scope variables `token` is parsed payload part, but in arguments `token` means whole JWT. Local-storage key is also `token` and meaning is the same as in argument.

I would change the naming to something like:
- `token` for JWT
- `session` for parsed payload part (as `session` is argument for navigation)

Reply:
Good call! I hope to refactor this to another component, when I have time. So this all will be encapsulated in something like `<LoginBoundary />` or similar.

I've actually changed  `token` in the methods to `rawToken` instead. That's because `token` really is a (parsed) token. Session sounds like the whole state shared between client and server (which is not much right now and is basically the token, but can grow).